### PR TITLE
blscfg: miscellaneous fixes

### PR DESCRIPTION
**Pull request by wjt**
I've been working on updating the Endless OS copy of this module based on the version in this repo, with the main goal of picking up the logic to explicitly sort the entries rather than relying on the dentries happening to be in the right order. We have some requirements that don't seem reasonable to push “upstream” to you, but here are some changes you might like.

If you're interested, the rest of the Endless OS branch is [here]( https://github.com/endlessm/grub/commits/T22742-rebase-onto-rhboot-grub-2.02-sb-plus-bls), with all the bls commits on top.

### CODE REVIEW

File: grub-core/commands/blscfg.c [hunk 1]
```diff
@@ -327,10 +327,26 @@ static int vercmp(const char * a, const char * b)
...
+  v0 = bls_get_val(e0, "version", NULL);
+  v1 = bls_get_val(e1, "version", NULL);
+
+  if (v0 && !v1)
+    return -1;
+  if (!v0 && v1)
+    return 1;
+
+  if ((r = vercmp(v0, v1)) != 0)
+    return r;
```
Comment (martinezjavier) [inline grub-core/commands/blscfg.c]: @wjt sorry that it caused an issue for you, I should had left sorting for the version field indeed.

Comment (wjt) [reply grub-core/commands/blscfg.c]: The only issue it caused is that I was confused why the menu was coming out in the wrong order, and on a hunch went and checked the ostree commit history :-)

**Comment by wjt** [comment]
Pushed 1 more compilation fix

**Comment by wjt** [comment]
And one more to fix `make distcheck`

**Comment by vathpela** [comment]
I've pushed all this to fedora-29.

**Pull request closed without merging**